### PR TITLE
Enable Go program generation

### PR DESCRIPTION
**Pull request by EvanBoyle**
Bump pulumi/pulumi to include generation of go programs.

I linked the AWS provider locally and the generated examples look pretty good. Definitely a few issues here or there that we're aware of, but good enough to ship some docs. https://github.com/pulumi/pulumi-aws/compare/evan/goExamples

Contributes to https://github.com/pulumi/pulumi-terraform-bridge/issues/216

### CODE REVIEW

File: gen/nodejs/testdata/test_meta_properties/index.ts [hunk 1]
```diff
@@ -6,5 +6,5 @@ const r1 = new aws.ec2.Instance("r1", {}, { timeouts: {
...
+const r2 = new aws.ec2.Instance("r2", {}, { ignoreChanges: ["ami", "arn", "associatePublicIpAddress", "availabilityZone", "cpuCoreCount", "cpuThreadsPerCore", "creditSpecification", "disableApiTermination", "ebsBlockDevices", "ebsOptimized", "ephemeralBlockDevices", "getPasswordData", "hibernation", "hostId", "iamInstanceProfile", "instanceInitiatedShutdownBehavior", "instanceState", "instanceType", "ipv6AddressCount", "ipv6Addresses", "keyName", "metadataOptions", "monitoring", "networkInterfaceId", "networkInterfaces", "outpostArn", "passwordData", "placementGroup", "primaryNetworkInterfaceId", "privateDns", "privateIp", "publicDns", "publicIp", "rootBlockDevice", "securityGroups", "sourceDestCheck", "subnetId", "tags", "tenancy", "userData", "userDataBase64", "volumeTags", "vpcSecurityGroupIds"] });
```
Comment (pgavlin) [inline gen/nodejs/testdata/test_meta_properties/index.ts:9]: These changes probably aren't necessary--the version of the AWS provider that is used in CI is likely different from the one used by your local machine. Same goes for the changes to `graph_test.go`.

**Comment by pgavlin** [comment]
The failing test is probably transient, but it does remind me that we should add support for Go and .NET in the test framework.

**Comment by EvanBoyle** [comment]
Opened https://github.com/pulumi/tf2pulumi/issues/190 to track adding go/dotnet. I will go ahead and merge.

**Comment by pgavlin** [comment]
Sounds good. We should fix #190 ASAP, as it's a good sanity check for example generation (i.e. passing tests in this repo are a strong signal that the example code gen is good enough).

**Comment by EvanBoyle** [comment]
Will try to take a look this sprint. Depending on the coverage (if it includes invokes) the go tests may not yet pass yet.

**Comment by pgavlin** [comment]
>  Depending on the coverage (if it includes invokes) the go tests may not yet pass yet.

If the Go code generator can't handle invokes, then it won't be able to generate docs for a sizable chunk of our SDK. We should make sure that the code generator hard-fails in that case, as otherwise we'll silently emit code that isn't copy-pastable.

**Pull request merged**